src/script_opt/ScriptOpt.cc

```diff
@@ -492,20 +492,50 @@ void analyze_scripts()
 	if ( ! analysis_options.activate )
 		return;
 
+	// The following tracks inlined functions that are also used
+	// indirectly, and thus should be compiled even if they were
+	// inlined.  We don't bother populating this if we're not inlining,
+	// since it won't be consulted in that case.
+	std::unordered_set<Func*> func_used_indirectly;
+
+	if ( inl )
+		{
+		for ( auto& f : funcs )
+			{
+			for ( const auto& g : f.Profile()->Globals() )
+				{
+				if ( g->GetType()->Tag() != TYPE_FUNC )
+					continue;
+
+				auto v = g->GetVal();
+				if ( ! v )
+					continue;
+
+				auto func = v->AsFunc();
+
+				if ( inl->WasInlined(func) )
+					func_used_indirectly.insert(func);
+				}
+			}
+		}
+
 	for ( auto& f : funcs )
 		{
-		if ( inl && inl->WasInlined(f.Func()) )
+		auto func = f.Func();
```

Comment:
`auto*` just for clarity that it's a pointer.

Reply:
I believe we use bare `auto` all over the place for values that are pointers - am I wrong about that?

Reply:
Huh, I thought it was the other way around. We definitely use bare `auto` for `IntrusivePtr`s, but they're not actually pointers.

Reply:
Ah I see.  I'm fine changing `auto` here to `auto*` ... however, IMHO the point of `auto` is to streamline code and make it (somewhat) resilient to things like changing bare pointers to `IntrusivePtr`'s, my druthers would be to leave this as it currently is. If that gives you heartburn then no problem, I'll go ahead and make the change.  (FWIW, I suspect I've done this elsewhere quite a bit ...)

Reply:
Hmmm I'm also remembering that a recent change I made to an API to use `IntrusivePtr` indeed worked fairly seamlessly because of the use - not in my own code - of `auto` for bare pointers.  So I think that idiom is employed generally in the code base.

Reply:
Sounds good to me to leave it alone then.